ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/remote/ConnectorUtils.java

```diff
@@ -245,4 +260,52 @@ public static SdkHttpFullRequest signRequest(
 
         return signer.sign(request, params);
     }
+
+    public static SdkHttpFullRequest buildSdkRequest(
+        Connector connector,
+        Map<String, String> parameters,
+        String payload,
+        SdkHttpMethod method
+    ) throws UnknownHostException {
+        String endpoint = connector.getPredictEndpoint(parameters);
+        Pattern pattern = Pattern.compile("(?:(\\w+)://)?((\\w+\\.)*\\w+)(?::(\\w+))?");
+        Matcher matcher = pattern.matcher(endpoint);
+        if (matcher.find()) {
+            String protocol = matcher.group(1);
```

Comment:
This seems bit brittle. What do you think about something like this:

```
String endpoint = "https://example.com:8080/path/to/resource?query=123";
            URL url = new URL(endpoint);

            // Getting protocol, host, and port
            String protocol = url.getProtocol();
            String host = url.getHost();
            int port = url.getPort(); // This will return -1 if the port is not explicitly specified in the URL
```

Reply:
Seems good, will test this approach.